lib/stages/HFBAccumulate.cpp

```diff
@@ -43,12 +46,50 @@ integrateHFBData::integrateHFBData(Config& config_, const std::string& unique_na
 
     out_buf = get_buffer("hfb_output_buf");
     register_producer(out_buf, unique_name.c_str());
+
+    // weight calculation is hardcoded, so is the weight type name
+    const std::string weight_type = "hfb_weight_type";
+    const std::string git_tag = get_git_commit_hash();
+    const std::string instrument_name =
+        config.get_default<std::string>(unique_name, "instrument_name", "chime");
+
+    std::vector<uint32_t> freq_ids;
+
+    // Get the frequency IDs that are on this stream, check the config or just
+    // assume all CHIME channels
+    // TODO: CHIME specific
+    if (config.exists(unique_name, "freq_ids")) {
+        freq_ids = config.get<std::vector<uint32_t>>(unique_name, "freq_ids");
+    } else {
+        freq_ids.resize(1024);
+        std::iota(std::begin(freq_ids), std::end(freq_ids), 0);
+    }
+
+    // Create the frequency specification
+    // TODO: CHIME specific
+    std::vector<std::pair<uint32_t, freq_ctype>> freqs;
+    std::transform(std::begin(freq_ids), std::end(freq_ids), std::back_inserter(freqs),
+                   [](uint32_t id) -> std::pair<uint32_t, freq_ctype> {
+                       return {id, {800.0 - 400.0 / 1024 * id, 400.0 / 1024}};
+                   });
```

Comment:
I think all of this CHIME-only code can be replaced with the new Telescope class functionality.